digdag-standards/src/main/java/io/digdag/standards/command/DockerCommandExecutor.java

```diff
@@ -107,29 +132,38 @@ private Process startWithDocker(Path projectPath, TaskRequest request, ProcessBu
             //    }
             //}
             //command.add("--env-file").add(envFile.toAbsolutePath().toString());
-            for (Map.Entry<String, String> pair : pb.environment().entrySet()) {
+            for (Map.Entry<String, String> pair : request.getEnvironments().entrySet()) {
                 command.add("-e").add(pair.getKey() + "=" + pair.getValue());
             }
 
             // image name
             command.add(imageName);
 
             // command and args
-            command.addAll(pb.command());
+            command.addAll(request.getCommandLine());
 
-            ProcessBuilder docker = new ProcessBuilder(command.build());
-            docker.redirectError(pb.redirectError());
-            docker.redirectErrorStream(pb.redirectErrorStream());
-            docker.redirectInput(pb.redirectInput());
-            docker.redirectOutput(pb.redirectOutput());
-            docker.directory(projectPath.toFile());
+            final ProcessBuilder pb = new ProcessBuilder(command.build());
+            pb.directory(workingDirectory.toFile());
+            pb.redirectErrorStream(true);
 
-            return docker.start();
+            return pb.start();
         }
         catch (IOException ex) {
             throw Throwables.propagate(ex);
         }
     }
+    private Path getAbsoluteWorkingDirectory(final CommandContext context, final CommandRequest request)
+    {
+        final Path workingDirectory;
+        if (!request.getWorkingDirectory().toString().isEmpty()) {
```

Comment:
why is this branch necessary...? Shouldn't we always use `context.getLocalProjectPath().resolve(request.getWorkingDirectory()).normalize()`?
as discussed offline before, this should throw an exception if created path is not under `context.getLocalProjectPath()` (this happens if `request.getWorkingDirectory()` is something like `..` or `../..`).

Reply:
Yes, I missed it. Thank you. Fixed https://github.com/treasure-data/digdag/pull/835/commits/691544a886f533ab54b68f22e0d1062d850aa765